lib/server.js

```diff
@@ -96,7 +96,10 @@ class S3Server {
      */
     cleanUp() {
         logger.info('server shutting down');
-        this.server.close(() => process.exit(0));
+        if (this.server) {
+            this.server.close(() => process.exit(0));
+        }
+        process.exit(0);
```

Comment:
This should be in an `else`, otherwise if there is a server attribute, the shutdown will not be graceful (what the callback is for).